src/utils/github/GitHubTicketStatusFetcher.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace staabm\PHPStanTodoBy\utils\github;
+
+use RuntimeException;
+use staabm\PHPStanTodoBy\utils\CredentialsHelper;
+use staabm\PHPStanTodoBy\utils\TicketStatusFetcher;
+
+use function array_key_exists;
+use function is_array;
+use function is_string;
+
+final class GitHubTicketStatusFetcher implements TicketStatusFetcher
+{
+    private const API_VERSION = '2022-11-28';
+
+    private string $owner;
+    private string $repo;
+    private ?string $accessToken;
+
+    /**
+     * @var array<string, ?string>
+     */
+    private array $cache;
+
+    public function __construct(string $owner, string $repo, ?string $credentials, ?string $credentialsFilePath)
+    {
+        $this->owner = $owner;
+        $this->repo = $repo;
+        $this->accessToken = CredentialsHelper::getCredentials($credentials, $credentialsFilePath);
+
+        $this->cache = [];
+    }
+
+    public function fetchTicketStatus(string $ticketKey): ?string
+    {
+        // trim "#"
+        $ticketKey = substr($ticketKey, 1);
```

Comment:
self explaining variant:

```suggestion
        $ticketKey = ltrim($ticketKey, '#');
```